x/service/types/message_add_service.go

```diff
@@ -30,5 +34,16 @@ func (msg *MsgAddService) ValidateBasic() error {
 	if msg.Service.Name == "" {
 		return ErrServiceMissingName
 	}
+	if err := ValidateComputeUnitsPerRelay(msg.Service.ComputeUnitsPerRelay); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidateComputeUnitsPerRelay makes sure the compute units per relay is a valid value
+func ValidateComputeUnitsPerRelay(computeUnitsPerRelay uint64) error {
+	if computeUnitsPerRelay == 0 {
```

Comment:
Wdyt of adding an upper bound to avoid potential overflows w/ all the multiplication?


If agreed, let's:
1. Add a local constant and set it to like 2^16
2. Add a unit test
3. Validate here

Reply:
Sounds good to me.